src/types/magnitude.rs

```diff
@@ -95,11 +95,22 @@ impl Magnitude {
         digits
     }
 
-    fn calculate_u64_digits(value: &u64) -> u64 {
-        match value {
-            0 => 1,
-            1 => 1,
-            i => (*i as f64).log10().ceil() as u64,
+    fn calculate_u64_digits(value: u64) -> u64 {
```

Comment:
🗺️ This helper method is used indirectly to determine if a given `Decimal` is in the range `[0, 1)`. It had a bug that caused the digit count for powers of ten to be off by one.